frontend/src/metabase/lib/analytics.ts

```diff
@@ -0,0 +1,99 @@
+import * as Snowplow from "@snowplow/browser-tracker";
+
+import { shouldLogAnalytics } from "metabase/env";
+import Settings from "metabase/lib/settings";
+
+export * from "./analytics-untyped";
+
+/**
+ * Tracks a schema event.
+ * IMPORTANT NOTE:
+ * Only use this function directly if `trackActionEvent` doesn't fit your needs.
+ * We want to use the generic structured action event as much as possible to avoid to create or update schemas.
+ * If you need to track event specific context that does not fit the structured action event, you can pass context entities
+ * that are designed in a reusable way (https://docs.snowplow.io/docs/understanding-your-pipeline/entities/)
+ */
+export const trackSchemaEvent = <TData extends object>(
+  /** The schema of the event. */
+  schema: string,
+  /** The version of the event. */
+  version: string,
+  /**  The data associated with the event. */
+  data: TData,
+  /** The context entities associated with the event. */
+  contextEntities?: Snowplow.SelfDescribingJson<Record<string, unknown>>[],
+): void => {
+  if (shouldLogAnalytics) {
+    const nameToLog =
+      schema === "structured_action" ? (data as any).name : (data as any).event;
+    // eslint-disable-next-line no-console
+    console.log(
```

Comment:
<img width="1290" alt="image" src="https://github.com/user-attachments/assets/8d49aafc-de1f-42f2-971e-d1f64e183a18">

Reply:
Nice!
That should nicely work for what we use it now. Could be that in the future there might be events that are structured differently (i.e. I would love to get away from this pattern of having a schema for a category of events and then specify `event` as a property -> this often leads to schemas that don't fit across the different `event`s). In those future cases there might not be a `name`, neither a `event` but we can optimize for it when it happens 👍